### DISCUSSION TITLE
Best result from compare_model vs. tune_model vs. calibrate_model vs. finalize_model

### DISCUSSION THREAD

**Discussion by marutilai** [pycaret/pycaret#3285] [General]
Which output can we finally use in production?

best_model = compare_model(n_select=1)
tuned_model = tune_model(best_model)
calibrated_model = calibrate_model()

I observed that running tune_model on the top result from compare_model gives better accuracy. 

1/ But if I call calibrate_model on top of this tuned_model then does that ruin the tuning?
2/ Do I call finalize model on top of tuned_model or calibrated_model?

**Comment by mc436572** [comment]
Mark and Wait for answer
